src/ci/tested_versions_utils.py

```diff
@@ -1,38 +1,180 @@
 from __future__ import annotations
 
 import os
+import re
 from contextlib import contextmanager
 from dataclasses import dataclass
-from typing import Tuple
+from typing import List, Union
+
+# Major, minor, patch and (non semver standard, suffix)
+_semanticVersionPattern = re.compile(r'(!)?(\d+).(\d+).(\d+)([^\s]*)(?:\s*#\s*(.*))?')
+_splitVersionFromCommentPattern = re.compile(r'(!)?([^\s]*)(?:\s*#\s*(.*))?')
+
+# This file implements a custom version parsing and sorting mechanism,
+# as `packaging.version` has strange behaviors that won't work for other
+# languages, like:
+#
+# ```
+# >>> from packaging.version import parse
+# >>> str(parse("1.2.4c"))
+# '1.2.4rc0'
+# ```
+#
+# ```
+# >>> from packaging.version import parse
+# >>> str(parse("1.2.4b"))
+# '1.2.4b0'
+# ```
+
+
+@dataclass(frozen=True)
+class NonSemanticVersion:
```

Comment:
Do we have such versions? I worry that maybe if a version will be identified as `NonSemanticVersion` it means that we have a bug at parsing the version. Maybe we should crash in this case instead of creating `NonSemanticVersion`?

Reply:
I believe we will find non-semantic versions in NPM and PIPY

Reply:
Actually, we _already_ have non semver around: https://github.com/lumigo-io/opentelemetry-python-distro/blob/770255e8bec507d5e186[%E2%80%A6]_opentelemetry/instrumentations/pymysql/tested_versions/pymysql